catalog/to-consul/syncer.go

```diff
@@ -218,18 +243,26 @@ func (s *ConsulSyncer) watchService(ctx context.Context, name string) {
 		case <-time.After(s.SyncPeriod):
 		}
 
+		// Set up query options
+		queryOpts := &api.QueryOptions{
+			AllowStale: true,
+		}
+		if s.EnableNamespaces {
+			// Sets the Consul namespace to query the catalog
+			queryOpts.Namespace = namespace
+		}
+
 		// Wait for service changes
 		var services []*api.CatalogService
 		err := backoff.Retry(func() error {
 			var err error
-			services, _, err = s.Client.Catalog().Service(name, s.ConsulK8STag, &api.QueryOptions{
-				AllowStale: true,
-			})
+			services, _, err = s.Client.Catalog().Service(name, s.ConsulK8STag, queryOpts)
 			return err
 		}, backoff.WithContext(backoff.NewExponentialBackOff(), ctx))
 		if err != nil {
 			s.Log.Warn("error querying service, will retry",
 				"service-name", name,
+				"service-namespace", namespace, // will be "" if namespaces aren't enabled
```

Comment:
Can we use `default` in every case and remove this special-case?

Reply:
Unfortunately not, as mentioned elsewhere.